core/services/keeper/upkeep_executer.go

```diff
@@ -207,24 +205,28 @@ func (executer *UpkeepExecuter) execute(upkeep UpkeepRegistration, headNumber in
 		runErrors = pipeline.RunErrors{null.StringFrom(errors.Wrap(err, "UpkeepExecuter: failed to construct upkeep txdata").Error())}
 	}
 
-	var etx bulletprooftxmanager.EthTx
-	ctxQuery, cancel := postgres.DefaultQueryCtx()
+	// postgres.DefaultQueryCtx() returns the same context with the same deadline.
+	// Just to avoid mixing store layer and service one the default context is used here.
+	txCtx, cancel := context.WithTimeout(context.Background(), time.Second*10)
 	defer cancel()
-	err = postgres.GormTransaction(ctxQuery, executer.orm.DB, func(dbtx *gorm.DB) (err error) {
-		etx, err = executer.orm.CreateEthTransactionForUpkeep(dbtx, upkeep, performTxData)
+
+	var etx bulletprooftxmanager.EthTx
+	err = executer.orm.WithTransaction(txCtx, func(ctx context.Context) error {
+		etx, err = executer.orm.CreateEthTransactionForUpkeep(ctx, upkeep, performTxData)
 		if err != nil {
 			return errors.Wrap(err, "failed to create eth_tx for upkeep")
 		}
 
 		// NOTE: this is the block that initiated the run, not the block height when broadcast nor the block
 		// that the tx gets confirmed in. This is fine because this grace period is just used as a fallback
 		// in case we miss the UpkeepPerformed log or the tx errors. It does not need to be exact.
-		err = executer.orm.SetLastRunHeightForUpkeepOnJob(dbtx, executer.job.ID, upkeep.UpkeepID, headNumber)
+		err = executer.orm.SetLastRunHeightForUpkeepOnJob(ctx, executer.job.ID, upkeep.UpkeepID, headNumber)
 		if err != nil {
 			return errors.Wrap(err, "failed to set last run height for upkeep")
 		}
 
-		_, err = executer.pr.InsertFinishedRun(dbtx, pipeline.Run{
+		// TODO: Pass context instead of *gorm.DB object. Here we are sure the context contains transaction object.
```

Comment:
could we just knock that out in this PR too?

Reply:
look at my last comment below